lib/Dialect/TritonGPU/IR/LinearLayoutConversions.cpp

```diff
@@ -314,15 +314,26 @@ LinearLayout sharedToLinearLayoutLeadingOffset(ArrayRef<int64_t> shape,
       bases2D.push_back({row, vec * ((row / perPhase) % maxPhase)});
     }
   }
+
+  // Replicate the tiles to fill out the inner dims of shapePerCTA.
+  for (int logRow = llvm::Log2_32(tileRows);
+       logRow < llvm::Log2_32(shapePerCTA[rowDim]); logRow++) {
+    bases2D.push_back({1 << logRow, 0});
+  }
+
```

Comment:
This is done automatically by `combineCtaCgaWithShape`.

Reply:
Yes. The order in which you declare the outDims of the layout is the order in which they are broadcasted. The first dim is the first one to be broadcasted and so on. The relevant code is here:
https://github.com/triton-lang/triton/blob/40f3945c6970d372355440a6c635f42932fc6987/lib/Tools/LayoutUtils.cpp#L135-L142
To get the correct broadcasting, you should just order the dimensions in the order you want them to broadcast.

The `combineCtaCgaWithShape` function will then broadacst and then transpose the dimensions to the standarised `dim0, dim1, ..., dimn-1`

Reply:
done based on your explanations